docs/architecture/adr-0001-app-setup-and-goals.md

```diff
@@ -0,0 +1,168 @@
+uses we envision for this project.
+
+## Specific Components
+
+### Summary
+
+TODO
+
+### New imagery from the outside world
+
+When new imagery becomes available from the outside world, this application will
+need to match it against models that are interested in that kind of imagery. This
+requires tracking some kind of metadata about the incoming imagery. We
+developed a fairly complex datamodel in Raster Foundry that tracks information about
+imagery, including component files for that imagery and band-level information.
+However, we are not a spec, and to the best of our knowledge no one relies on the
+Raster Foundry datamodel to describe what imagery updates look like.
+
+Instead, the API will demand that incoming imagery adheres to the [STAC electro-optical](https://github.com/radiantearth/stac-spec/tree/master/extensions/eo)
+spec extension. There are several reasons to prefer using STAC for a datamodel.
+These include tapping into ongoing work in the geospatial community to standardize
+data exchange, making validation easier for anyone hoping to rely on this API,
+and reusing our own work around a type-safe STAC datamodel in `geotrellis-server`.
+This choice also commits us to using Scala as the backend language. Using anything
+else would require us to rewrite data modeling work in another language, which is
+time consuming and would quickly get either ahead of or behind whatever it's
+supposed to be in sync with.
+
+### Running models against new imagery
+
+`POST`ing a new image to the API will cause that image to be matched against
+available models. A `model` is some function from raster to raster, raster to
+geojson, or raster to number. For simplicity, we'll start with the raster to geojson
+case, since it will save us some complexity on figuring out what kinds of
+contracts we need to enforce for the other kinds of models and is sufficient for
+object detection and chip classification tasks. `Model`s know what sorts of imagery
+are acceptable for them to obtain meaningful results:
+
+- an array of bandwidth requirements
+- a minimum spatial resolution
+- an area of interest
+
+We can determine this information for any incoming imagery that adheres to the
+`eo` spec using band information.
+
+`Model`s also need to know their own requirements to run. In particular, they
+need sufficient information to be able to fill in a `SubmitJob` request for `AWS`
+Batch. This includes resource requirements, a docker container accessible from
+within the hosting AWS account, a compute environment name, and a command to run.
+We require `model`s to carry this information to avoid having to infer it or
+guess at it based on heuristics.
+
+Once we identify that a `model` should run against an image, we'll kick off an
+AWS Batch job with the model's parameters. The `model`'s command _must_ accept a
+STAC item with the `eo` extension as its last parameter.
+
+`Model`s will be renamed to something less overloaded for developer sanity at
```

Comment:
I think there is a missing datamodel here - `AOI` -- I think an `AOI` would encompass the spatial and image filters and have one or more models to kick off if incoming STAC data passes those filters.